src/Fixer/PhpUnit/PhpUnitDataProviderNameFixer.php

```diff
@@ -133,13 +134,23 @@ protected function applyPhpUnitClassFix(Tokens $tokens, int $startIndex, int $en
                 continue;
             }
 
+            // Skip abstract data providers.
+            if (null !== $this->findAbstractToken($dataProviderAnalysis->getNameIndex(), $tokens)) {
+                continue;
+            }
+
             $usageIndex = $dataProviderAnalysis->getUsageIndices()[0];
             if (substr_count($tokens[$usageIndex]->getContent(), '@dataProvider') > 1) {
                 continue;
             }
 
             $testNameIndex = $tokens->getNextTokenOfKind($usageIndex, [[T_STRING]]);
 
+            // Skip abstract tests.
+            if (null !== $this->findAbstractToken($testNameIndex, $tokens)) {
```

Comment:
Only the provider name is renamed, to be consistent, this condition should not be applied.

Reply:
This condition prevents cases where the test is declared abstract and the provider name is referenced by a concrete implementation of the test in child classes.

Reply:
That is true for non-abstract tests as well, thus it should be better consistent. It is good practise to not redeclare/copy paste the phpdoc in override methods anayway.

Reply:
I agree. I'm just trying to prevent all the possible cases where we know this fixer will create unintended changes.